src/profiler.cpp

```diff
@@ -1647,6 +1652,104 @@ void Profiler::dumpText(Writer& out, Arguments& args) {
     }
 }
 
+/*
+ * If the value has been seen already, return its idx in `values`. If not, update the map,
+ * add the value to `values`, and return the index.
+ */
+static inline u64 getOrSetIndex(
+        std::unordered_map<std::string, u64>& idx_map,
+        std::vector<std::string>& values,
```

Comment:
There is no need to fill a vector right here - it's always possible to reconstruct a vector from the map later when it is needed.

Reply:
Sure, but can we obtain an ordered list of values without sorting the content of the map?

Reply:
Yes, just iterate over `{key, value}` pairs in the map and use `value` as an array index for `key`.
Here is [Java equivalent](https://github.com/async-profiler/async-profiler/blob/master/src/converter/one/convert/Index.java#L42).

Reply:
Ah right, didn't think about that. Thanks

Reply:
e9837729ca6146eb5c9d4f54f6662af307537b1c